transport/transport.go

```diff
@@ -0,0 +1,36 @@
+/*
+Package transport defines interfaces for the Tektite simple RPC framework - this is used for sending RPCs between
+different Tektite nodes
+ */
+package transport
+
+/*
+Server is the interface implemented by a transport server
+ */
+type Server interface {
+	// RegisterHandler is called to register a RequestHandler with the specified id
+	RegisterHandler(handlerID int, handler RequestHandler) bool
+	// Address returns the address of the Server
+	Address() string
+	// Stop stops the server - releasing all resources
+	Stop() error
+}
+
+// ResponseWriter is called to send a response back to the client, or an error
+type ResponseWriter func(response []byte, err error) error
+
+// RequestHandler is implemented on the server to handle a request and return a response
+type RequestHandler func(request []byte, responseBuff []byte, responseWriter ResponseWriter) error
+
+// ConnectionFactory returns a Connection that connects to the specified server address
+type ConnectionFactory func(address string) (Connection, error)
+
+/*
+Connection is the interface implemented by a client connection
+ */
+type Connection interface {
+	// SendRPC sends a request to the server handler and returns a response, or an error
+	SendRPC(handlerID int, request []byte) ([]byte, error)
```

Comment:
What is the handler in this context? What would a client populate here

Reply:
The handler is the thing on the server side that handles the request.
The client just sends the request and receives the response, or an error

Reply:
What client does this represent? A remote one, right? How would the client know the ID of the handler on the server side?

Do we use things like Request ID?

Reply:
The handler ids are known at compile time. Think of them like the ApiKey in the Kafka protocol - there's one for each operation.

Reply:
> What client does this represent? A remote one, right?

The client doesn't know where the handler is - it could be local or remote.